subprojects/execution/src/main/java/org/gradle/internal/execution/fingerprint/FileCollectionFingerprinter.java

```diff
@@ -38,7 +37,7 @@
     /**
      * Creates a fingerprint from the snapshot of a file collection.
      */
-    CurrentFileCollectionFingerprint fingerprint(FileSystemSnapshot snapshot, @Nullable FileCollectionFingerprint previousFingerprint);
+    CurrentFileCollectionFingerprint fingerprint(FileCollection files, @Nullable FileCollectionFingerprint previousFingerprint);
```

Comment:
Feels like we are taking a step back here. We should instead move towards requiring a snapshot all the time, no?

Reply:
That is what we could already do now on `master`, there is no reason to wait for 8.0.

So for me, the plan would be to be able to register a fingerprinting strategy, and then obtain that strategy via a spec. So `FileCollectionFingerprinter` would go away. Should we try to pursue this?

Reply:
Let's do that!